beekeeper-core/src/main/java/com/expediagroup/beekeeper/core/config/PrometheusConfigFactory.java

```diff
@@ -0,0 +1,64 @@
+/**
+ * Copyright (C) 2019-2020 Expedia, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.expediagroup.beekeeper.core.config;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.stereotype.Component;
+
+import io.micrometer.prometheus.PrometheusConfig;
+
+@Component
+@ConfigurationProperties(prefix = "prometheus")
+public class PrometheusConfigFactory {
+
+  private static String DEFAULT_PREFIX = "beekeeper";
+
+  private String prefix;
+
+  public String getPrefix() {
+    return prefix;
+  }
+
+  public void setPrefix(String prefix) {
+    this.prefix = prefix;
+  }
+
+  public PrometheusConfig newInstance() {
+    if (prefix == null) {
```

Comment:
NABD but I think you could assign `prefix` to the default on line 29 and then not have this if check here.